playbooks/roles/ocp-config/README.md

```diff
@@ -11,32 +11,51 @@ Requirements
+| chronyconfig.allow          | no       | ""                                    | List of network cidr (X.X.X.X/Y) allowed to sync with bastion configured as NTP server |
+| dhcp_shared_network         | no       |                                       | Flag to update DHCP server work on a shared network. (Neither ACK nor NACK unknown clients) |
+| cni_network_provider        | no       | OVNKubernetes                         | Sets the default Container Network Interface (CNI) network provider for the cluster |
+| cni_network_mtu             | no       |                                       | MTU value to assign to the CNI network. Recommended values for OpenshiftSDN: <NIC MTU> - 50; OVNKubernetes: <NIC MTU> - 100 |
+| cluster_network_cidr        | no       | 10.128.0.0/14                         | Network (in CIDR) used for the pod networks.
+| cluster_network_hostprefix  | no       | 23                                    | The subnet prefix length to assign to each individual node. (netmask in CIDR format)
+| service_network             | no       | 172.30.0.0/16                         | Network (in CIDR) used for the service network.
+| rhcos_pre_kernel_options    | no       | []                                    | List of day-1 kernel options for RHCOS nodes eg: ["rd.multipath=default","root=/dev/disk/by-label/dm-mpath-root"] |
+| luks.enabled                | no      | false                                 | Set it true if you prefer to enable LUKS in ocp deployment |
+| luks.config                 | yes      | []                                    | List of tang servers and thumbprint to apply |
+| thumbprint                  | yes      |                                       | Thumbprint of tang server to be added in luks_config |
+| url                         | yes      |                                       | URL of tang server to be added in luks_config |
+| luks.filesystem_device      | yes      | /dev/mapper/root                      | Set the Path of device to be luks encrypted |
+| luks.format                 | yes      | xfs                                   | Set the Format of the FileSystem to be luks encrypted |
+| luks.wipeFileSystem         | yes      | true                                  | Configures the FileSystem to be wiped |
+| luks.device                 | yes      | /dev/disk/by-partlabel/root           | Set the Path of luks encrypted partition |
+| luks.label                  | yes      | luks-root                             | Set the value for user label of luks encrpted partition |
+| luks.options                | yes      | ["--cipher", "aes-cbc-essiv:sha256"]  | Set List of luks options for the luks encryption |
+| luks.wipeVolume             | yes      | true                                  | Configures the luks encrypted partition to be wiped |
+| luks.name                   | yes      | root                                  | Set the value for the user label of Filesystem to be luks encrypted |
```

Comment:
I see you have documented these variables are required and provided defaults but where are these defaults defined in code? Also, if you have defined defaults then it should be optional and not required.

Reply:
If they are required then a validation in code is necessary.

Reply:
Updated defaults for LUKS in defaults/main/main.yaml and also updated the required value in documentation to 'no'